src/pages/ReportSettingsPage.js

```diff
@@ -107,23 +108,46 @@ class ReportSettingsPage extends Component {
             },
         };
 
+        this.roomNameInputRef = null;
+
         this.state = {
             newRoomName: this.props.report.reportName,
             errors: {},
         };
 
-        this.validateAndRenameReport = this.validateAndRenameReport.bind(this);
+        this.setRoomNameInputRef = this.setRoomNameInputRef.bind(this);
+        this.resetToPreviousName = this.resetToPreviousName.bind(this);
+        this.validateAndUpdatePolicyRoomName = this.validateAndUpdatePolicyRoomName.bind(this);
     }
 
-    validateAndRenameReport() {
-        if (!this.validate()) {
-            return;
+    /**
+     * Set the room name input ref
+     *
+     * @param {Element} el
+     */
+    setRoomNameInputRef(el) {
+        this.roomNameInputRef = el;
+    }
+
+    /**
+     * When the user dismisses the error updating the policy room name,
+     * reset the report name to the previously saved name and clear errors.
+     */
+    resetToPreviousName() {
+        this.setState({newRoomName: this.props.report.reportName});
+
+        // Reset the input's value back to the previously saved report name
+        if (this.roomNameInputRef) {
+            this.roomNameInputRef.value = this.props.report.reportName.replace(CONST.POLICY.ROOM_PREFIX, '');
```

Comment:
I wasn't sure about this part at first, but then remembered React Native inputs are more performant when they are uncontrolled.

Still, I'm not sure if modifying the value like this is 100% correct - I vaguely recall there being issues on native related to assigning the value directly. Did we test this out there?

Reply:
I will test on iOS and Android to see if it works there. I did a quick google search for how to clear a text input and went off of [this SO](https://stackoverflow.com/questions/45249807/clear-react-native-textinput), so it seems like the accepted way to do it.

Reply:
Woops, that SO is about clearing the text input, which isn't exactly what I want to do in this case. You're right that this approach doesn't work. When I tested on iOS I was unable to enter anything after the value was directly set.

Should I use [setNativeProps](https://reactnative.dev/docs/direct-manipulation#setnativeprops-to-edit-textinput-value) or should I make this a controlled input, or something else?

Reply:
Looks like we use `setNativeProps` in a couple other places, and it works well, so I'll go with that.
https://github.com/Expensify/App/blob/782ea4a7c1350b36519bf4dff2512e0fa37b07e8/src/pages/home/report/ReportActionCompose.js#L375
https://github.com/Expensify/App/blob/1bfae69c3dce03d017270cc9838c480610e925c1/src/components/Form.js#L167

Reply:
Controlled components have some performance issues on native so I usually recommend the uncontrolled. But if this were a web only app I'd say to use controlled components as I prefer working with them (e.g. clearing an input just requires the parent setting a state value to `''` which is more intuitive IMO). `setNativeProps` looks like the only option to me.